packages/core/src/lib/lit-core.ts

```diff
@@ -945,7 +971,25 @@ export class LitCore {
       body: JSON.stringify(data),
     };
 
-    return sendRequest(url, req, requestId);
+    let abortTimeout: ReturnType<typeof setTimeout>;
+    if (this.config.nodeRequestTimeout) {
+      // TODO: Use AbortSignal.timeout once we update to TS >=4.9.5. It considers request active time instead of elapsed time
+      // AbortSignal.timeout(this.config.nodeRequestTimeout)
+      const abortController = new AbortController();
+      req.signal = abortController.signal;
+      abortTimeout = setTimeout(
+        () => abortController.abort('Request to node timed out'),
+        this.config.nodeRequestTimeout
+      );
+    }
```

Comment:
This timeout also applies for lit action or any other request made to the nodes. I can change it to just the handshake ones, or make it configurable afterwards to give consumers more flexibility